DataFormats/Provenance/src/ProductProvenance.cc

```diff
@@ -14,23 +14,22 @@ namespace {
 namespace edm {
   ProductProvenance::ProductProvenance() : branchID_(), parentageID_() {}
 
-  ProductProvenance::ProductProvenance(BranchID bid) : branchID_(std::move(bid)), parentageID_() {}
+  ProductProvenance::ProductProvenance(BranchID bid) : branchID_(bid), parentageID_() {}
 
   ProductProvenance::ProductProvenance(BranchID bid, ParentageID edid)
-      : branchID_(std::move(bid)), parentageID_(std::move(edid)) {}
+      : branchID_(bid), parentageID_(std::move(edid)) {}
 
   ProductProvenance::ProductProvenance(BranchID bid, std::vector<BranchID> const& parents)
-      : branchID_(std::move(bid)), parentageID_() {
+      : branchID_(bid), parentageID_() {
     Parentage p;
     p.setParents(parents);
     parentageID_ = p.id();
     ParentageRegistry::instance()->insertMapped(p);
   }
 
-  ProductProvenance::ProductProvenance(BranchID bid, std::vector<BranchID>&& parents)
-      : branchID_(std::move(bid)), parentageID_() {
+  ProductProvenance::ProductProvenance(BranchID bid, std::vector<BranchID>&& parents) : branchID_(bid), parentageID_() {
     Parentage p;
-    p.setParents(std::move(parents));
+    p.setParents(parents);
```

Comment:
Looks like we need a new setParents function in Parentage which passes the item by value.

Reply:
I think `std:move` is unnecessary here because `parents` is already a `&&`.

Reply:
I checked and `setParents` takes a `const &` which would require a copy. I'm going to change  `Parentage::setParents` so moves will work properly with it.